src/libstore/build.cc

```diff
@@ -799,6 +799,9 @@ class DerivationGoal : public Goal
     /* Whether this is a fixed-output derivation. */
     bool fixedOutput;
 
+    /* Whether this is a content adressed derivation */
+    bool contentAddressed = false;
```

Comment:
@regnat Sorry I have not responded to this PR more thoroughly yet.

First thing I wanted to point out is I made https://github.com/NixOS/nix/pull/3418 so that we might talk about the various independent-ish aspects of derivations, rather than keeping around multiple bools of which only 1 should be true.